developer_manual/app/schema.rst

```diff
@@ -1,3 +1,141 @@
+===================
+Database Migrations
+===================
+
+.. sectionauthor:: Thomas Müller <thomas.mueller@tmit.eu>
+
+ownCloud uses migration steps to perform any kind of changes between versions. In most cases these changes are changes
+to the database schema but it is not only limited to them.
+
+Therefore we support three kinds of migration steps: simple, sql and schema.
+
+There are currently 3 types of migrations supported:
+
+- simple: this allows general migration steps to be executed - pretty similar to the `repair steps <https://doc.owncloud.org/api/classes/OCP.Migration.IRepairStep.html>`_
+- sql: this allows you to add a list of sql commands to be executed
+- schema: migration steps of this kind allow various schema migration operations.
+
+Starting with ownCloud 10 this is the preferred way to perform any kind of migrations and is enabled by default within core.
```

Comment:
this being the whole migration section - the old way is the xml stuff at the end of the file.